### PR TITLE
test: reuse connection create and process request tests

### PR DESCRIPTION
**Pull request by jakubkoci**
Just small PR adding ConnectionService tests I removed in the previous PR. I also added some checks for the OOB role and state.

### CODE REVIEW

File: packages/core/src/modules/connections/services/ConnectionService.ts [hunk 1]
```diff
@@ -74,12 +75,14 @@ export class ConnectionService {
...
-  public async protocolCreateRequest(
+  public async createRequest(
     outOfBandRecord: OutOfBandRecord,
     config: ConnectionRequestParams
   ): Promise<ConnectionProtocolMsgReturnType<ConnectionRequestMessage>> {
-    // TODO check oob role is sender
-    // TODO check oob state is await-response
+    this.logger.debug(`Create message ${ConnectionRequestMessage.type} start`, outOfBandRecord)
+    outOfBandRecord.assertRole(OutOfBandRole.Receiver)
+    outOfBandRecord.assertState([OutOfBandState.Initial, OutOfBandState.PrepareResponse])
```
Comment (TimoGlastra) [inline packages/core/src/modules/connections/services/ConnectionService.ts:84]: In what case is it initial and what case is it prepareResponse?

File: packages/core/src/modules/oob/repository/OutOfBandRecord.ts [hunk 2]
```diff
@@ -67,14 +67,20 @@ export class OutOfBandRecord extends BaseRecord<TagsBase> {
       .reduce((acc, curr) => [...acc, ...curr], [])
   }
 
+  public assertRole(expectedRole: OutOfBandRole | undefined) {
```
Comment (TimoGlastra) [inline packages/core/src/modules/oob/repository/OutOfBandRecord.ts:70]: Why can expectedRole be undefined?

**Review by TimoGlastra — APPROVED** [review]

File: packages/core/src/modules/oob/repository/OutOfBandRecord.ts (hunk 2 above)
Comment (jakubkoci) [reply packages/core/src/modules/oob/repository/OutOfBandRecord.ts]: I copy-pasted from `connectionRecord` where I put it and left it by accident I guess :) Updated in both places.

File: packages/core/src/modules/connections/services/ConnectionService.ts (hunk 1 above)
Comment (jakubkoci) [reply packages/core/src/modules/connections/services/ConnectionService.ts]: It didn't make sense. I updated it.

Comment (jakubkoci) [reply packages/core/src/modules/connections/services/ConnectionService.ts]: I wasn't using initial and setting it directly to one of prepare/await response. So the check should be more strict. But I updated oob module to set oob state to `initial` in `createInvitation` and then update it in `acceptInvitation`. The method `recieveInvitation` stays same, setting `await-response` directly to a new oob record.

**Comment by jakubkoci** [comment]
@TimoGlastra I did updates based on your suggestions, can please check if it makes sense?

**Pull request merged**